docs/README.md

```diff
@@ -1,20 +1,201 @@
-# appfwk
-DUNE DAQ Application Framework Repository 
+# appfwk: The DUNE DAQ Application Framework
 
-***
-* [Step-by-step instructions for creating your own DAQModule package under v2.0.0](Step-by-step-instructions-for-creating-your-own-DAQModule-package-under-v2.0.0) (_under construction_)
-* [Step-by-step instructions for creating your own DAQModule package under v1.1.0](Step-by-step-instructions-for-creating-your-own-DAQModule-package-under-v1.1.0)
-* [Step-by-step instructions for creating your own DAQModule package under v1.0.0](Step-by-step-instructions-for-creating-your-own-DAQModule-package-under-v1)
-* [Step-by-step instructions for creating your own DAQModule package](Step-by-step-instructions-for-creating-your-own-DAQModule-package)
+## Overview
 
-***
+**In a single sentence:** 
 
-* [How to write your DAQModule](How-to-write-your-DAQModule)
-* [Application Framework Contributor's Guide](Contributors-Guide)
+appfwk consists of a generic DAQ application (`daq_application`) which can be configured to run different types of DAQ modules performing various DAQ-related tasks, where DAQ modules are user-developed implementations of appfwk's `DAQModule` interface class. 
+
+**In a bit more detail:**
+
+appfwk provides the scaffolding on which all DUNE DAQ software processes can be developed. The running DAQ typically consists of multiple distinct processes assigned various tasks: filtering data, requesting it, saving it to storage, etc. There are many different types of process, some of which may not even have been conceived of yet, and it would be cumbersome to recompile multiple different types of process across many packages every time one wanted to change the behavior of the DAQ. To solve this problem, the approach that's been taken is to have a standard DUNE DAQ software process [`daq_application`](Daq-Application.md) which can be configured at runtime by Run Control in order to perform some particular function in the DAQ. 
+
+Drilling down a bit deeper, the way that `daq_application`s can be configured so flexibly is that they are designed to have DAQ modules embedded in them, where DAQ modules are units of code designed to perform specific tasks and which pass data between each other by means of containers known as "queues". These specific tasks can vary widely; they include [producing fake data for testing purposes](https://github.com/DUNE-DAQ/readout/blob/develop/plugins/FakeCardReader.hpp), [putting data into long term storage](https://github.com/DUNE-DAQ/dfmodules/blob/develop/plugins/HDF5DataStore.hpp), and so forth. DAQ modules will typically execute user-defined functions when receiving standard transitions from Run Control: conf, start, etc. appfwk provides the `DAQModule` base class which users should derive their DAQ module class from in their own packages.  
```

Comment:
The HDF5 example is wrong. HDF5DataStore is not a `DAQModule`, here it creates confusion. Use `DataWriter`.